commit aa042bd0f416edfb0eedcac6be6ba90b1c1c161c
Author: Albert Astals Cid <aacid@kde.org>
Date:   Sat Aug 18 18:36:30 2012 +0200

    Close the url on shell destruction
    
    Otherwise we end up in a busy loop on part deletion if there are pending requests
    Should not happen but if this fixes it don't see the need to lose time investigating
    why given the number of todo things

diff --git a/shell/shell.cpp b/shell/shell.cpp
index df40f7d..8b99ce4 100644
--- a/shell/shell.cpp
+++ b/shell/shell.cpp
@@ -119,7 +119,11 @@ void Shell::delayedOpen()
 
 Shell::~Shell()
 {
-    if ( m_part ) writeSettings();
+    if ( m_part )
+    {
+        writeSettings();
+        m_part->closeUrl();
+    }
     m_part = 0; // It is deleted by the KPart/QObject machinery
     if ( m_args )
         m_args->clear();
